### PR TITLE
merge consecutive string literals

### PR DESCRIPTION
**Pull request by andrewchambers**
Dunno if i should be freeing the old nodes.

### CODE REVIEW

**Comment by oridb** [comment]
Mostly LGTM. Can you rebase and add a test?

**Comment by andrewchambers** [comment]
done, added some docs too

File: doc/lang.txt [hunk 1]
```diff
@@ -192,6 +192,13 @@ TABLE OF CONTENTS:
 
             eg: "foo\"bar"
 
+        Multiple consecutive string literals are implicitly merged to create
+        a single combined string literal. To allow a string literal to span
+        across multiple lines, the new line characters must be be escaped.
```
Comment (phase) [inline doc/lang.txt]: `be be escaped` -> `be escaped`

Comment (oridb) [inline doc/lang.txt]: Thanks! Pushed a fix.

**Pull request merged**